fix(lxc): multiple issues during container import

CaptaiNiveau opened the pull request:
### Contributor's Note
<!--- 
Please mark the following items with an [x] if they apply to your PR.
Leave the [ ] if they are not applicable, or if you have not completed the item.
--->
- [ ] I have added / updated documentation in `/docs` for any user-facing features or additions.
- [ ] I have added / updated acceptance tests in `/fwprovider/tests` for any new or updated resources / data sources.
- [x] I have ran `make example` to verify that the change works as expected.

<!---
You can find more information about coding conventions and local testing in the [CONTRIBUTING.md](https://github.com/bpg/terraform-provider-proxmox/blob/main/CONTRIBUTING.md) file.

If you are unsure how to run `make example`, see [Deploying the example resources](https://github.com/bpg/terraform-provider-proxmox?tab=readme-ov-file#deploying-the-example-resources) section in README.
--->

<!--
*IF* your code contains breaking changes make sure to add `!` to the end of commit type, e.g.:
```
    feat(vm)!: add support for new feature 
```
Also, uncomment the section just below, and add a description of the breaking change. 
--->

<!---
#### ⚠ BREAKING CHANGES

>>> Put your description here <<<
--->

### Proof of Work
<!--- 
Please add screenshots, logs, or other relevant information that demonstrates the change works as expected.
--->
Previously, a configuration generated from an existing container looked like this:
```tf
resource "proxmox_virtual_environment_container" "testcontainer" {
  ...
  timeout_update      = null
  unprivileged        = null
  vm_id               = null
  ...
```

This led to tofu wanting to replace the container based on the just generated config `unprivileged = null` (which results in false) and the difference to the imported state `unprivileged = null`. This is also an issue with `unprivileged = true|false`, as it's compared against null.

Now, the property `unprivileged` is imported correctly:
```tf
resource "proxmox_virtual_environment_container" "testcontainer" {
  ...
  timeout_update      = null
  unprivileged        = true
  vm_id               = null
  ...
```

<!--- Please keep this note for the community --->
### Community Note

- Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request
<!--- Thank you for keeping this note for the community --->

<!--- If your PR fully resolves and should automatically close the linked issue, use Closes. Otherwise, use Relates --->
Closes #1787

<!--- Release note for [CHANGELOG](https://github.com/bpg/terraform-provider-proxmox/blob/main/CHANGELOG.md) will be created automatically using the PR's title, update it accordingly. --->

CaptaiNiveau commented:
Three more things: 
- I just rebased this, forgot to test again with the new commits, but hopefully that's not an issue
- The mode of imported device passthroughs to containers is `""`, while Proxmox' default value is `0660`. The empty string leads to a parsing error: `Error: invalid value for mode (Octal access mode)`. I changed the default value locally and it works as expected. Should I add it to this PR or a different one?
- During testing, I ran into a few issues with the Proxmox configuration. The documentation is mostly very good, but I ran into a few issues. Mainly the variables and the example in `setup-proxmox-for-testing.md`. Should I open another PR for this?

Thank you for your work, this repo was very nice to work with. I'm just getting into Terraform/OpenTofu and I've also never worked with go, the docs helped a lot.

CaptaiNiveau commented:
Oh, I just realized that I wrote `fix(lxc)` instead of `fix(container)` in the commit message (and missed the "This fixes ..." in the end, but you got that one already ^^)

bpg commented:
Hey @CaptaiNiveau 👋🏼 

Thanks for the PR, outside contributions are always welcome and greatly appreciated!! ❤️ 

>     * I just rebased this, forgot to test again with the new commits, but hopefully that's not an issue

👍🏼 
 
>     * The mode of imported device passthroughs to containers is `""`, while Proxmox' default value is `0660`. The empty string leads to a parsing error: `Error: invalid value for mode (Octal access mode)`. I changed the default value locally and it works as expected. Should I add it to this PR or a different one?

Let's go ahead and include that in the same PR. Maybe rename it to something like `fix(lxc): multiple issues during container import` or similar. If you make it a separate commit, it'll be easier to review.   

>     * During testing, I ran into a few issues with the Proxmox configuration. The documentation is mostly very good, but I ran into a few issues. Mainly the variables and the example in `setup-proxmox-for-testing.md`. Should I open another PR for this?
> 

Yeah, that probably warrants a separate PR. Feel free to open one!

> Thank you for your work, this repo was very nice to work with. I'm just getting into Terraform/OpenTofu and I've also never worked with go, the docs helped a lot.

Thank you for the kind words, that means a lot to me! ❤️ 
Feel free to open PRs / issues for any other improvements you may think of.

bpg commented:
Also you'd need to sing your commits, pls take a look [here](https://github.com/bpg/terraform-provider-proxmox/pull/1867/checks?check_run_id=39639024712) for more details.

bpg commented:
> Oh, I just realized that I wrote `fix(lxc)` instead of `fix(container)` in the commit message (and missed the "This fixes ..." in the end, but you got that one already ^^)

Yeah, I've updated that to match the convention

CaptaiNiveau commented:
Hi, I changed the commit message to include sign-off, but I'm not sure how to get it onto here. I also added the default passthrough mode, which already got on here.

I tried force pushing to my repo that I forked, but that didn't work. How do I replace these commits?

CaptaiNiveau commented:
Ah okay, I finally got it. I initially used `--force-with-lease` instead of `--force`. Just `--force` works as expected. Now the commits are signed and should work as expected, the commit to change the default value of passthrough mode is included, too.

Also, I rebased the commits again on main during my experimentation. I haven't done many PRs yet - rebasing on main is probably not the best idea without retesting. I reran my local setup with the current version (using `go install .` and `~/.terraformrc`) and it worked as expected. `make example` however failed, since there's already modifications from previous runs like an existing user. How do you get a clean proxmox setup for testing - take a snapshot and reset it every time or something similar?

CaptaiNiveau commented:
Also, most of the tests I ran using `make test` result in `? ... [no test files]` locally, some are `ok`. How do I use them properly?

Maybe they should also be expanded by a test case for these additions, not sure what your coverage looks like.

bpg commented:
> Also, I rebased the commits again on main during my experimentation. I haven't done many PRs yet - rebasing on main is probably not the best idea without retesting. I reran my local setup with the current version (using `go install .` and `~/.terraformrc`) and it worked as expected. `make example` however failed, since there's already modifications from previous runs like an existing user. How do you get a clean proxmox setup for testing - take a snapshot and reset it every time or something similar?

`make example-delete` _should_ delete all dangling resources from a previous `make example` run. Sometimes it depends when exactly the previous run failed, some resources might be "locked", or the local TF state might not be properly updated. Then you have to go and delete them manually 😞

> Also, most of the tests I ran using make test result in ? ... [no test files] locally, some are ok. How do I use them properly?

`make test` doesn't really do much besides checks for API message formats parsing and some other data transformation validation. There are more involved ["acceptance tests"](https://github.com/bpg/terraform-provider-proxmox/blob/main/CONTRIBUTING.md#acceptance-tests), however they are fragile and could be tricky to set up. 
TBH I don't expect external contributors going through the hassle of setting them up, so I'm running them myself as part of PR review 😅

> Maybe they should also be expanded by a test case for these additions, not sure what your coverage looks like.

Ya, I'll take a look.


Also, [historically, I'm using `lxc`](https://github.com/bpg/terraform-provider-proxmox/blob/main/CONTRIBUTING.md#commit-message-conventions) in the commit summary to indicate the container-related changes :)

bpg reviewed (approved):
I've also set the schema default to "0660" to prevent unexpected plan diffs on re-apply (i.e., the new value `"0660"` ends up in the state, but it isn’t in the config). Alternatively, we could declare `mode` as a computed attribute, but that might complicate things during cloning. Also updated the acceptance tests.

LGTM! 🚀 Thank you for contribution!

bpg commented:
@all-contributors please add @CaptaiNiveau for code
